analytics/track.go

```diff
@@ -0,0 +1,80 @@
+package analytics
+
+import (
+	"bytes"
+	"encoding/json"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+const poolSize = 10
+const bufferSize = 100
+
+// Tracker ...
+type Tracker interface {
+	Enqueue(eventName string, properties map[string]interface{})
+	Wait()
+}
+
+type tracker struct {
+	jobs             chan *bytes.Buffer
+	waitGroup        *sync.WaitGroup
+	client           Client
+	sharedProperties map[string]interface{}
+}
+
+// NewDefaultTracker ...
+func NewDefaultTracker(shared map[string]interface{}) Tracker {
+	return NewTracker(make(chan *bytes.Buffer, bufferSize), &sync.WaitGroup{}, poolSize, NewDefaultClient(), shared)
+}
+
+// NewTracker ...
+func NewTracker(jobs chan *bytes.Buffer, waitGroup *sync.WaitGroup, size int, client Client, sharedProperties map[string]interface{}) Tracker {
+	t := tracker{jobs: jobs, waitGroup: waitGroup, client: client, sharedProperties: sharedProperties}
+	t.init(size)
+	return &t
+}
+
+// Enqueue ...
+func (t tracker) Enqueue(eventName string, properties map[string]interface{}) {
```

Comment:
why not pass an `Event` here? by doing so later we could easier alter the Event struct (default formats and fields can be handled with a constructor for the Event struct)

Reply:
might Event should be an interface, which can be JSONMarshalled, and shared fields can be set on it.

Reply:
I wouldn't set the shared properties on the Event, other than that it sounds ok